realm/realm-library/src/main/java/io/realm/internal/Collection.java

```diff
@@ -421,12 +431,12 @@ public boolean deleteLast() {
         if (observerPairs.isEmpty()) {
             nativeStartListening(nativePtr);
         }
-        CollectionObserverPair<T> collectionObserverPair = new CollectionObserverPair<T>(observer, listener);
+        CollectionObserverPair<T> collectionObserverPair = new CollectionObserverPair<>(observer, listener);
```

Comment:
I take it back.  There are 10s of incompatibilities, in the UTs.  There is no way this is gonna compile w/1.6, without some boring work.  Out of scope.  I fixed the ~20 in the main code and put the `sourceCompatibility` back to default.